### PR TITLE
Feat/ipfs connection

### PR DESCRIPTION
**Pull request by 0xArdi**
## Proposed changes

This PR adds a connection for handling IPFS uploads and downloads, along with the required changes to the IPFSBehaviour.
This is a **breaking** change. Breaking changes:
- ipfs connection & protocol (valory/ipfs:0.1.0) needs to be added to your `aea-config.yaml`, if they utilize `valory/abstract_round_abci`.
- ipfs dialogues and handler need to be specified.
- `send_to_ipfs` and `get_from_ipfs` are now moved to `BaseBehaviour` and are generators. 

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [x] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: packages/valory/connections/ipfs/__init__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Scaffold of a connection."""
```
Comment (0xArdi) [inline packages/valory/connections/ipfs/__init__.py:20]: TODO

File: packages/valory/skills/abstract_round_abci/behaviour_utils.py [hunk 2]
```diff
@@ -369,85 +374,77 @@ def __handle_stop_iteration(self) -> None:
...
     """Behaviour for interactions with IPFS."""
 
     def __init__(self, **kwargs: Any):
         """Initialize an `IPFSBehaviour`."""
         super().__init__(**kwargs)
-        self.ipfs_enabled = False
-        # If params are not found `AttributeError` will be raised. This is fine, because something will have gone wrong.
-        # If `ipfs_domain_name` is not specified for the skill, then we get a `None` default.
-        # Therefore, `IPFSBehaviour` will be disabled.
-        domain = getattr(self.params, "ipfs_domain_name", None)  # pylint: disable=E1101
```
Comment (DavidMinarsch) [inline packages/valory/skills/abstract_round_abci/behaviour_utils.py:401]: Please update the models.py and remove the attribute there as it is no longer used anywhere now afaik

File: packages/valory/skills/abstract_round_abci/skill.yaml [hunk 3]
```diff
@@ -99,6 +101,9 @@ handlers:
   tendermint:
     args: {}
     class_name: TendermintHandler
+  ipfs:
```
[the remark this answers is not in the document]
Comment (0xArdi) [reply packages/valory/skills/abstract_round_abci/skill.yaml]: done [here](https://github.com/valory-xyz/open-autonomy/pull/1721/commits/f757f0b2f6a47b704fda98cec86a92563a026142)

File: autonomy/fsm/scaffold/templates/components.py [hunk 4]
```diff
@@ -299,6 +299,9 @@ class HANDLERS:
     from packages.valory.skills.abstract_round_abci.handlers import (
```
Comment (DavidMinarsch) [inline autonomy/fsm/scaffold/templates/components.py:299]: you also need to update the scaffold of skill.yaml

File: packages/valory/protocols/ipfs/protocol-spec.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
+---
```
Comment (DavidMinarsch) [inline packages/valory/protocols/ipfs/protocol-spec.yaml:1]: This should be moved into `README.md` as this is currently how our generate all logic handles it. See open-aea

**Review by DavidMinarsch — APPROVED** [review]

File: packages/valory/skills/test_ipfs_abci/payloads.py [hunk 6]
```diff
@@ -34,7 +35,8 @@ def __str__(self) -> str:
         return self.value
 
 
+@dataclass(frozen=True)
 class DummyPayload(BaseTxPayload, ABC):
```
Comment (DavidMinarsch) [inline packages/valory/skills/test_ipfs_abci/payloads.py:39]: please remove ABC

**Review by DavidMinarsch — APPROVED** [review]

File: packages/valory/agents/register_reset_recovery/aea-config.yaml [hunk 7]
```diff
@@ -87,14 +89,23 @@ public_id: valory/p2p_libp2p_client:0.1.0
...
+  public_key: ${P2P_PUBLIC_KEY:str:02d3a830c9d6ea1ae91936951430dee11f4662f33118b02190693be835359a9d77}
+  save_path: .certs/acn_cosmos_9005.txt
+- identifier: acn
+  ledger_id: ethereum
+  message_format: '{public_key}'
+  not_after: '2023-01-01'
+  not_before: '2022-01-01'
+  public_key: ${P2P_PUBLIC_KEY:str:02e741c62d706e1dcf6986bf37fa74b98681bc32669623ac9ee6ff72488d4f59e8}
+  save_path: .certs/acn_cosmos_9006.txt
+is_abstract: true
```
Comment (Adamantios) [inline packages/valory/agents/register_reset_recovery/aea-config.yaml:111]: This is a merge error.

Comment (Adamantios) [reply packages/valory/agents/register_reset_recovery/aea-config.yaml:111]: Addressed on #1747.

**Pull request merged**